mako/compat.py

```diff
@@ -99,19 +99,31 @@ def octal(lit):
 
 
 if py3k:
-    from importlib import machinery
-
-    def load_module(module_id, path):
-        return machinery.SourceFileLoader(module_id, path).load_module()
-
+    from importlib import machinery, util
+
+    if hasattr(util, 'module_from_spec'):
+        # Python 3.5+
+        def load_module(module_id, path):
+            spec = util.spec_from_file_location(module_id, path)
+            module = util.module_from_spec(spec)
+            sys.modules[module_id] = module
+            spec.loader.exec_module(module)
```

Comment:
do we want to have:

            del sys.modules[module_id]

here?   the idea is to not pollute sys.modules.  seems to have all tests pass.

Reply:
Oh dear, I left `sys.modules[module_id] = module` in. I didn't mean to do that.
`module_from_spec`/`exec_module` doesn't put the module in `sys.modules` in the first place.

I'removed the line and adjusted the test to check that `sys.modules` isn't polluted.